File: src/tests/Markify.Fixtures/Projects/Source/Struct/StructMethods.vb
Public Partial Structure FooType
    Sub Method()
	End Sub

    Public Sub PublicMethod()
	End Sub

    Function IntMethod() As Integer
		Return 1
	End Function

    Sub WithParametersMethod(foo As Integer, ByRef bar As Integer, ByVal foobar As Integer)
	End Sub

    Private Partial Sub PartialMethod()
    End Sub

    Function SingleGenericMethod(Of T)(Optional foo As T = Nothing) As T
		Return CType(Nothing, T)
	End Function

    Function MultiGenericMethod(Of T As IList, Y As {IDisposable, IEnumerable})(foo As T, bar As y) As T
		Return CType(Nothing, T) 
	End Function
End Structure

File: src/tests/Markify.Fixtures/Projects/Source/Struct/StructMethods.cs
public partial struct FooType
{
    void Method() { }

    public void PublicMethod() { }

    int IntMethod() { return 1; }

    partial void PartialMethod();

    void WithParametersMethod(int foo, ref int bar, out int foobar) { foobar = 2; }

    void WithNoNameParameterMethod(__arglist) { }

    T SingleGenericMethod<T>(T foo = default(T)) { return default(T); }

    T MultiGenericMethod<T, Y>(T foo, Y bar)
        where T : IList
        where Y : IDisposable, IEnumerable
    {
        return default(T);
    }

    int BodyMethod(int foo) => 1;
}